core/parentchain/block-import-dispatcher/src/lib.rs

```diff
@@ -37,15 +37,76 @@ pub mod triggered_dispatcher;
 #[cfg(feature = "mocks")]
 pub mod trigger_parentchain_block_import_mock;
 
-use error::Result;
-use std::vec::Vec;
+use crate::triggered_dispatcher::TriggerParentchainBlockImport;
+use core::marker::PhantomData;
+use error::{Error, Result};
+use std::{sync::Arc, vec::Vec};
 
 /// Trait to dispatch blocks for import into the local light-client.
-pub trait DispatchBlockImport {
-	type SignedBlockType;
-
+pub trait DispatchBlockImport<SignedBlockType> {
 	/// Dispatch blocks to be imported.
 	///
 	/// The blocks may be imported immediately, get queued, delayed or grouped.
-	fn dispatch_import(&self, blocks: Vec<Self::SignedBlockType>) -> Result<()>;
+	fn dispatch_import(&self, blocks: Vec<SignedBlockType>) -> Result<()>;
+}
+
+/// Wrapper struct for the actual dispatchers. Allows to define one global type for
+/// both dispatcher without the need to change the global variable when switching
+/// the dispatcher type. It also allows for empty dispatchers, for use cases that
+/// do not need block syncing for a specific parentchain type.
+pub struct BlockImportDispatcher<TriggeredDispatcher, ImmediateDispatcher, SignedBlockType> {
+	triggered_dispatcher: Option<Arc<TriggeredDispatcher>>,
+	immediate_dispatcher: Option<Arc<ImmediateDispatcher>>,
+	_phantom: PhantomData<SignedBlockType>,
+}
```

Comment:
Hmm, I am uncertain if I agree with that approach. Usually, a downstream implementor chooses either offchain-worker or sidechain globally for his project. So I think it is actually better to change one line of code instead of introducing all this complexity.

Empty dispatcher could also just be a trait implementation on something that should not dipsatch anything at all.

Reply:
Although, it is not too much complexity. So I don't have a strong opinion here. :)

Reply:
Our CI compiles all modes, so it's not just about downstream implementors. So either we define for every Worker Mode a different parentchain type or we use a WorkerMode dependent initialization, e.g. introduce compiler flags.
I'm fine with all of these solutions.

Reply:
> Empty dispatcher could also just be a trait implementation on something that should not dipsatch anything at all.

I think that one is dangerous. I'd rather throw an error if an empty dispatcher is called, because then it will be noticed straight away. If no dispatcher should be called, then I believe it's better to not call the `dispatch` at all.